### PR TITLE
[v5.x] remove AOT costing code

### LINKED ISSUE #792: Get rid of AOT costing code in SigmaCompiler
### Problem
Currently (as of v4.x, v5.0), the `SigmaCompiler.compile` method is implemented using IRContext.doCosting method.
The method builds cost graph along with the calc graph.
In v5.x the cost graph is no longer necessary.

### Solution

The construction of cost graph can be sliced off the doCosting implementation.

### PR DESCRIPTION
**Pull request by aslesarenko**
Closes #792 
In this PR:
- [x] Removed v4.x interpreter implementation and all the code used by it
- [x] Removed AOT costing implementation and related code of graph IR
- [x] Removed code used by old costing
- [x] Removed all code that became unused after the above were removed
- [x] Added more ScalaDoc

Note:
1) There is no new code, all new lines are actually old code moved from other places.
2) All test remained unchanged (modulo renaming and removing of some properties)

### CODE REVIEW

File: sigmastate/src/main/scala/sigmastate/eval/GraphBuilding.scala [hunk 1]
```diff
@@ -34,8 +43,321 @@ trait GraphBuilding extends SigmaLibrary { IR: IRContext =>
   import SigmaDslBuilder._
   import SigmaProp._
   import WOption._
+
+  /** Should be specified in the final cake */
+  val builder: sigmastate.lang.SigmaBuilder
   import builder._
```
Comment (aslesarenko) [inline sigmastate/src/main/scala/sigmastate/eval/GraphBuilding.scala:50]: This is not new code, it is moved from other traits of the IR cake.

**Review by kushti — CHANGES_REQUESTED** [review]
Will re-check `CBigInt` and `CGroupElement` during the second round (reminder for self)

File: sigmastate/src/main/scala/org/ergoplatform/SigmaConstants.scala [hunk 2]
```diff
@@ -32,11 +32,11 @@ object SigmaConstants {
     "Registers count should not be greater than provided value") {
   }
 
-  object MaxPropositionBytes extends SizeConstant[Int](Sized.SizePropositionBytesMax.dataSize.toInt, 5,
+  object MaxPropositionBytes extends SizeConstant[Int](4096 /*4K*/, 5,
```
[the remark this answers is not in the document]
Comment (aslesarenko) [reply sigmastate/src/main/scala/org/ergoplatform/SigmaConstants.scala:35]: The doc for each constant is in the description. Added ScalaDoc for SizeConstant class.

File: sigmastate/src/main/scala/org/ergoplatform/SigmaConstants.scala [hunk 3]
```diff
@@ -32,11 +32,11 @@ object SigmaConstants {
...
     "Max length of Box.propositionBytes collection") {
   }
 
-  object MaxBoxSizeWithoutRefs extends SizeConstant[Int](Sized.SizeBoxBytesWithoutRefsMax.dataSize.toInt, 6,
+  object MaxBoxSizeWithoutRefs extends SizeConstant[Int](MaxBoxSize.value - (CryptoConstants.hashLength + 2/*size of Short*/), 6,
```
[the remark this answers is not in the document]
Comment (aslesarenko) [reply sigmastate/src/main/scala/org/ergoplatform/SigmaConstants.scala:39]: same as above

File: sigmastate/src/main/scala/sigmastate/eval/GraphBuilding.scala [hunk 4]
```diff
@@ -34,8 +43,321 @@ trait GraphBuilding extends SigmaLibrary { IR: IRContext =>
...
+  }
+
+  type RColl[T] = Ref[Coll[T]]
+  type ROption[T] = Ref[WOption[T]]
+
+  private val CBM      = CollBuilderMethods
+  private val SigmaM   = SigmaPropMethods
+  private val SDBM     = SigmaDslBuilderMethods
+
+  object AnyOf {
```
[the remark this answers is not in the document]
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/eval/GraphBuilding.scala:88]: fixed.

File: sigmastate/src/main/scala/sigmastate/interpreter/InterpreterContext.scala [hunk 5]
```diff
@@ -122,6 +122,6 @@ trait InterpreterContext {
     *
     * @see sigmastate.eval.Evaluation
     */
-  def toSigmaContext(isCost: Boolean, extensions: Map[Byte, AnyValue] = Map()): sigma.Context
+  def toSigmaContext(extensions: Map[Byte, AnyValue] = Map()): sigma.Context
```
[the remark this answers is not in the document]
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/interpreter/InterpreterContext.scala:123]: fixed

**Review by kushti — APPROVED** [review]

**Pull request merged**